realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -585,6 +622,7 @@ private void emitCopyOrUpdateMethod(JavaWriter writer) throws IOException {
                 .beginControlFlow("if (rowIndex != TableOrView.NO_MATCH)")
                     .emitStatement("realmObject = new %s()", Utils.getProxyClassName(className))
                     .emitStatement("realmObject.realm = realm")
+                    .emitStatement("realmObject.updateColumnInfo(realm.getColumnInfo(%s.class))", className)
```

Comment:
There is a RealmObjectProxy interface you could use instead?
Also, my brain just understood why you need to have an `updateColumnIndices` on the proxy classes, you need some way to parse information to all the accessors - DOH

I would use the RealmObjectProxy interface instead, we should avoid adding methods to RealmObject if at all possible because it impact what the user can do.